kalamine/layout.py

```diff
@@ -138,9 +159,9 @@ def __init__(self, filepath: Path) -> None:
         """Import a keyboard layout to instanciate the object."""
 
         # initialize a blank layout
-        self.layers = [{}, {}, {}, {}, {}, {}]
-        self.dk_set = set()
-        self.dead_keys = {}  # dictionary subset of DEAD_KEYS
+        self.layers: Dict[Layer, Dict[str, str]] = {layer: {} for layer in Layer}
```

Comment:
This change is weird but seemed necessary for consistency.

Reply:
I wrote that too late yesterday. Thinking about it this morning under my shower, it is no longer weird. `self.layers` used to be a list accessed as a `Dict[Layer, Dict[str, str]]` because you can convert a `Layer` into an `int` and use as index of a list.

Reply:
This line is very self-explanatory, but what’s the point in using a Dict rather than a List here ? Doesn’t that defeat the idea of a `Layer` enum ?

And if we follow that route, I think that `for layer in Layer` could be more readable, maybe `for index in Layer` or something.

Reply:
Indeed this is roughly saying we want an `Enum` instead of `IntEnum`. My modification makes things easier for typechecking (mypy certainly complained about the original version) and I think it's more readable. I don't think the ordering implied by the use of `IntEnum` is relevant here. One can argue that shift layers come after their regular version but I don't see why ODK and ALTGR would be ordered.

Reply:
Makes sense. I guess I tried to apply an old-school mental model I got from C here.